### PR TITLE
Update aarch64 makefiles

### PR DESCRIPTION
**Pull request by kubanrob**
This PR modifies the Makefiles to select the right sources for aarch64. It includes 3 commits from @marcrittinghaus.

This PR applies on top of PR #9 and should be used together with #6.

### CODE REVIEW

**Comment by kubanrob** [comment]
`__set_thread_area.s` needs to be removed from makefile when merged into https://github.com/unikraft/lib-musl/pull/9. Will be removed.

**Comment by kubanrob** [comment]
Update: Rebase to latest state of thread_support (PR #9)

File: Makefile.uk [hunk 1]
```diff
@@ -95,13 +97,13 @@ $(LIBMUSL)/src/internal/version.h:
...
-UK_PREPARE += $(LIBMUSL)/arch/$(ARCH)/bits/alltypes.h
+UK_PREPARE += $(LIBMUSL)/arch/$(MUSL_ARCH)/bits/alltypes.h
 UK_PREPARE += $(LIBMUSL)/src/internal/version.h
 UK_PREPARE += $(LIBMUSL_BUILD)/.prepared
 
-LIBMUSL_CLEAN += $(LIBMUSL)/arch/$(ARCH)/bits/alltypes.h
+LIBMUSL_CLEAN += $(LIBMUSL)/arch/$(MUSL_ARCH)/bits/alltypes.h
 LIBMUSL_CLEAN += $(LIBMUSL)/src/internal/version.h
-LIBMUSL_CLEAN += $(LIBMUSL_BUILD)/include
+#LIBMUSL_CLEAN += $(LIBMUSL_BUILD)/include
```
Comment (razvanvirtan) [inline Makefile.uk:106]: Why do we remove this file? Is the reason connected to the goal of this commit?

File: Makefile.uk.musl.string [hunk 2]
```diff
@@ -91,6 +91,10 @@ LIBMUSL_STRING_SRCS-y += $(LIBMUSL)/src/string/arm/memcpy.c|arm
 LIBMUSL_STRING_SRCS-y += $(LIBMUSL)/src/string/arm/memcpy_le.S|arm
 LIBMUSL_STRING_SRCS-y += $(LIBMUSL)/src/string/memmove.c|arm
 LIBMUSL_STRING_SRCS-y += $(LIBMUSL)/src/string/memset.c|arm
+else ifeq (arm64,$(CONFIG_UK_ARCH))
+LIBMUSL_STRING_SRCS-y += $(LIBMUSL)/src/string/aarch64/memcpy.s|aarch64
+LIBMUSL_STRING_SRCS-y += $(LIBMUSL)/src/string/aarch64/memset.s|x86_64
```
Comment (razvanvirtan) [inline Makefile.uk.musl.string]: Shouldn't this be `......|aarch64`?

File: Makefile.uk (hunk 1 above)
Comment (razvanvirtan) [inline Makefile.uk]: What is the purpose of removing this file? Is it connected to the goal of this commit?

**Comment by razvanvirtan** [comment]
Hi, Robert!
Besides the inline comments, there is a minor typo in https://github.com/unikraft/lib-musl/pull/7/commits/c3a54479bd51610b47032c35fff76ffc3fec0a53 (`configuration of`).

**Review by razvanvirtan — COMMENTED** [review]
Hi, Robert!
Besides the inline comments, there is a minor typo in https://github.com/unikraft/lib-musl/pull/7/commits/c3a54479bd51610b47032c35fff76ffc3fec0a53 (`configuration of`).

**Review by eduardvintila — APPROVED** [review]
Hi, @kubanrob. Apart from what @razvanvirtan said, everything looks good to me.

Reviewed-by: Eduard Vintilă <eduard.vintila47@gmail.com>

File: Makefile.uk (hunk 1 above)
Comment (kubanrob) [reply Makefile.uk]: This is a directory, which breaks the clean rule. So probably should be removed properly, and maybe should live in a different PR.

File: Makefile.uk.musl.string (hunk 2 above)
Comment (kubanrob) [reply Makefile.uk.musl.string]: Yes, will update.

**Comment by kubanrob** [comment]
**Update:** I removed the change to `LIBMUSL_CLEAN += $(LIBMUSL_BUILD)/include` and merged the two commits updating the sublibrary makefiles into one.

For the fixing the clean build target I will open a new PR, as commenting out / removing that line is not enough.

**Comment by razvanvirtan** [comment]
Thank you, @kubanrob!
I also double checked the Makefile.uk.* files with your script and everything looks good.

Reviewed-by: Razvan Virtan <virtanrazvan@gmail.com>

File: Makefile.uk (hunk 1 above)
Comment (kubanrob) [reply Makefile.uk]: The new PR can be found here: PR #17

**Comment by razvand** [comment]
@razvanvirtan , please also approve the PR with the approve button.

**Review by razvanvirtan — APPROVED** [review]
Thank you, @kubanrob!
I also double checked the Makefile.uk.* files with your script and everything looks good.

Reviewed-by: Razvan Virtan <virtanrazvan@gmail.com>

**Comment by razvand** [comment]
Thanks, @kubanrob . This works. I need to see whether I can approve it, or we should first integrated PR #9 and then you do a rebase.

**Comment by kubanrob** [comment]
**Update**: rebased onto staging

**Review by razvand — APPROVED** [review]
Thanks, @kubanrob 

Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request closed without merging**